extensions/wikia/CommunityPage/scripts/CommunityPageBenefitsModal.js

```diff
@@ -57,12 +67,31 @@ define('CommunityPageBenefitsModal',
 		}
 
 		/**
-		 * CreateComponent callback that finally shows modal
-		 * and binds submit action to Done button
-		 * One of sub-tasks for getting modal shown
+		 * CreateComponent callback that finally shows modal.
+		 * Bind tracking events.
+		 * One of sub-tasks for getting modal shown.
 		 */
 		function processInstance(modalInstance) {
 			modalInstance.show();
+
+			// Send tracking event for modal shown
+			track({
+				action: tracker.ACTIONS.IMPRESSION,
+				label: 'benefits-modal-shown'
+			});
+
+			// Bind tracking on elements with data-track attribute
+			modalInstance.$element.find('[data-track]').on('mousedown', function (e) {
+				track({label: $(e.target).data('track')});
+			});
+
+			// Bind tracking modal close
+			modalInstance.bind('close', function () {
```

Comment:
modalInstance is custom Wikia object that doesn't provide .on method
https://github.com/Wikia/app/blob/2d75c4bedd478d3746aaeb48e10b0440c7fa6359/resources/wikia/ui_components/modal/js/modal.js#L325-L325